### PR TITLE
fixes #1 initial terraform to deploy ranger

### LINKED ISSUE #1: terraform module to deploy ranger admin and usersync
As sysops,
I wan't to create terraform module intial version to deploy ranger admin and usersync services on ECS,

### PR DESCRIPTION
**Pull request by rpoluri**
Aurora database for storing ranger configs and audit logs
 ranger admin service HA configuration with sticky sessions.
 ranger usersync service to sync ldap users and groups from Active Directory
 read database master password from vault
 route53 dns entries for ranger admin & database
 *solr based auditing is not included in intial commit.

### CODE REVIEW

File: db.tf [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  db_subnet_group_name = "${aws_db_subnet_group.ranger.name}"
+  publicly_accessible  = false
+  tags                 = "${var.apiary_tags}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "ranger_database" {
```
Comment (pradeepbhadani) [inline db.tf]: just curious to know the benefit of R53 entry on RDS endpoint?

without R53 entry, it makes hards to detect or guess the name of RDS endpoint.

File: iam.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "task_exec_managed" {
```
Comment (pradeepbhadani) [inline iam.tf]: rename resource to `ranger_task_exec_policy`. Makes more readable

File: output.tf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/**
```
Comment (pradeepbhadani) [inline output.tf]: remove the blank file?
We should try to use more of data resources instead of outputs where possible.

File: ranger.tf [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+  port              = "6080"
+  protocol          = "HTTP"
+
+  default_action {
+    target_group_arn = "${aws_lb_target_group.ranger_admin_tg.arn}"
+    type             = "forward"
+  }
+}
+
+data "template_file" "ranger_usersync" {
```
Comment (pradeepbhadani) [inline ranger.tf]: move this to different file?

File: db.tf (hunk 1 above)
Comment (rpoluri) [reply db.tf]: so that we can use this to configure ranger plugins for DB audit provider

File: ranger.tf (hunk 4 above)
Comment (rpoluri) [reply ranger.tf]: done.

File: iam.tf (hunk 2 above)
Comment (rpoluri) [reply iam.tf]: done

File: db.tf (hunk 1 above)
Comment (pradeepbhadani) [reply db.tf]: can't we use the actual endpoint generated by RDS?

Comment (rpoluri) [reply db.tf]: we can, but if ever redeploy authorization, then you need go and update all datalakes with new DNS entries for RDS.

alias makes it easy to decouple deployments.
if this is too concerning for your environment, we can add flag to make this optional.

Comment (pradeepbhadani) [reply db.tf]: okay. let's go with R53 entry for now. if we need to make it optional, we can handle that later.

**Review by pradeepbhadani — APPROVED** [review]
looks good

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,3 +3,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] - 2018-08-28
```
Comment (massdosage) [inline CHANGELOG.md]: Remove date until you're ready to tag and release it.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,3 +3,12 @@ All notable changes to this project will be documented in this file.
...
+### Added
+- initial terraform: https://github.com/ExpediaInc/apiary-authorization/issues/1
```
Comment (massdosage) [inline CHANGELOG.md]: Markdown syntax for hyperlink please

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,3 +3,12 @@ All notable changes to this project will be documented in this file.
...
+- Aurora database for storing ranger configs and audit logs
+- ranger admin service HA configuration with sticky sessions.
+- ranger usersync service to sync ldap users and groups from Active Directory
+- read database master password from vault
+- route53 dns entries for ranger admin & database
+- *solr based auditing is not included in intial commit.
```
Comment (massdosage) [inline CHANGELOG.md]: Remove *

**Review by mroark1m — APPROVED** [review]
looks good.

**Pull request merged**